packages/authorization-process/src/routers/AuthorizationRouter.ts

```diff
@@ -118,6 +118,56 @@ const authorizationRouter = (
         }
       }
     )
+    .get(
+      "/clientsWithKeys",
+      authorizationMiddleware([
+        ADMIN_ROLE,
+        SECURITY_ROLE,
+        M2M_ROLE,
+        SUPPORT_ROLE,
+      ]),
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+        try {
+          const { name, userIds, consumerId, purposeId, kind, offset, limit } =
+            req.query;
+          const clients = await authorizationService.getClients({
+            filters: {
+              name,
+              userIds: userIds?.map(unsafeBrandId<UserId>),
+              consumerId: unsafeBrandId(consumerId),
+              purposeId: purposeId
+                ? unsafeBrandId<PurposeId>(purposeId)
+                : undefined,
+              kind,
+            },
+            authData: req.ctx.authData,
+            offset,
+            limit,
+            logger: ctx.logger,
+          });
+          return res
+            .status(200)
+            .json({
+              results: clients.results.map((client) =>
+                clientToApiClient(client, {
+                  includeKeys: true,
+                  showUsers: ctx.authData.organizationId === client.consumerId,
+                })
+              ),
+              totalCount: clients.totalCount,
+            })
+            .end();
+        } catch (error) {
+          const errorRes = makeApiProblem(
+            error,
+            getClientsErrorMapper,
```

Comment:
I would declare a dedicated mapper like:
`getClientsWithKeysErrorMapper = getClientsErrorMapper`

Otherwise it would seem like an error

Reply:
Updated